tests/integration/main.rs

```diff
@@ -0,0 +1,197 @@
+#![allow(non_snake_case)]
+
+/// Tests against `.orc` and `.jsn.gz` in the official test suite (`orc/examples/`)
+use std::fs::File;
+use std::io::Read;
+
+use pretty_assertions::assert_eq;
+
+use arrow::array::StructArray;
+use arrow::record_batch::RecordBatch;
+use datafusion_orc::arrow_reader::ArrowReaderBuilder;
+
+/// Checks parsing a `.orc` file produces the expected result in the `.jsn.gz` path
+fn test_expected_file(name: &str) {
+    let dir = env!("CARGO_MANIFEST_DIR");
+    let orc_path = format!("{}/tests/integration/data/{}.orc", dir, name);
+    let jsn_gz_path = format!("{}/tests/integration/data/expected/{}.jsn.gz", dir, name);
+    let f = File::open(orc_path).expect("Could not open .orc");
+    let builder = ArrowReaderBuilder::try_new(f).unwrap();
+    let orc_reader = builder.build();
+    let total_row_count = orc_reader.total_row_count();
+
+    // Read .orc into JSON objects
+    let batches: Vec<RecordBatch> = orc_reader.collect::<Result<Vec<_>, _>>().unwrap();
+    let objects: Vec<serde_json::Value> = batches
+        .into_iter()
+        .map(|batch| -> StructArray { batch.into() })
+        .flat_map(|array| {
+            arrow_json::writer::array_to_json_array(&array)
+                .expect("Could not serialize convert row from .orc to JSON value")
+        })
+        .collect();
+
+    // Read expected JSON objects
+    let mut expected_json = String::new();
+    flate2::read::GzDecoder::new(&File::open(jsn_gz_path).expect("Could not open .jsn.gz"))
+        .read_to_string(&mut expected_json)
+        .expect("Could not read .jsn.gz");
+
+    let objects_count = objects.len();
+
+    // Reencode the input to normalize it
+    let expected_lines = expected_json
+        .split('\n')
+        .filter(|line| !line.is_empty())
+        .map(|line| {
+            serde_json::from_str::<serde_json::Value>(line)
+                .expect("Could not parse line in .jsn.gz")
+        })
+        .map(|v| {
+            serde_json::to_string_pretty(&v).expect("Could not re-serialize line from .jsn.gz")
+        })
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    let lines = objects
+        .into_iter()
+        .map(|v| serde_json::to_string_pretty(&v).expect("Could not serialize row from .orc"))
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    if lines.len() < 1000 {
+        assert_eq!(lines, expected_lines);
+    } else {
+        // pretty_assertions consumes too much RAM and CPU on large diffs,
+        // and it's unreadable anyway
+        assert_eq!(lines[0..1000], expected_lines[0..1000]);
+        assert!(lines == expected_lines);
```

Comment:
I wonder if we can parse the expected JSON into Arrow first with arrow_json then compare on RecordBatches

Assuming the schema inference works in our favour :thinking:

Reply:
That would avoid the issue of different decimal representation.

However, it makes the tests a little unreliable, as they wouldn't detect data lost both by `arrow_json` and `datafusion_orc`. Probably not a big deal

Reply:
That's a fair point. I guess we're a bit handicapped by the expected data being in JSON form which can make it harder for us to be rigorous. I'll raise an issue for exploring ways around this :+1: